lib/active_storage/service/azure_service.rb

```diff
@@ -0,0 +1,69 @@
+require 'azure/storage'
+require "active_support/core_ext/numeric/bytes"
+
+class ActiveStorage::Service::AzureService < ActiveStorage::Service
+  attr_reader :client, :path, :blobs, :container
+
+  def initialize(path:, storage_account_name:, storage_access_key:, container:)
+    @client = Azure::Storage::Client.create(storage_account_name: storage_account_name, storage_access_key: storage_access_key)
+    @blobs = client.blob_client
+    @container = container
+    @path = path
+  end
+
+  def upload(key, io, checksum: nil)
+    instrument :upload, key, checksum: checksum do
+      begin
+        blobs.create_block_blob(container, key, io, content_md5: checksum)
+      rescue Azure::Core::Http::HTTPError
+        raise ActiveStorage::IntegrityError
+      end
+    end
+  end
+
+  def download(key)
+    instrument :download, key do
+      _, io = blobs.get_blob(container, key)
+      io
```

Comment:
For compatibility with the other services, this method should read from the IO in chunks and yield each chunk if a block is given, or read the IO completely and return its data.